subsys/bluetooth/host/keys.c

```diff
@@ -325,7 +325,8 @@ void bt_keys_clear(struct bt_keys *keys)
 
 	LOG_DBG("%s (keys 0x%04x)", bt_addr_le_str(&keys->addr), keys->keys);
 
-	if ((keys->flags & BT_KEYS_ID_CONFLICT) != 0) {
+	if (IS_ENABLED(CONFIG_BT_ID_AUTO_SWAP_MATCHING_BONDS) &&
```

Comment:
add `nrf-squash! [nrf noup] bluetooth: host: Add support for bonding with same peer` to commit message

Reply:
Done